cluster/store/schema.go

```diff
@@ -298,14 +298,26 @@ func (s *schema) getTenants(class string) ([]*models.Tenant, error) {
 	// Read tenants using the meta lock guard
 	var res []*models.Tenant
 	f := func(_ *models.Class, ss *sharding.State) error {
-		res = make([]*models.Tenant, len(ss.Physical))
-		i := 0
-		for tenant := range ss.Physical {
-			res[i] = &models.Tenant{
-				Name:           tenant,
-				ActivityStatus: entSchema.ActivityStatus(ss.Physical[tenant].Status),
+		if len(tenants) == 0 {
+			res = make([]*models.Tenant, len(ss.Physical))
+			i := 0
+			for tenant := range ss.Physical {
+				res[i] = &models.Tenant{
+					Name:           tenant,
+					ActivityStatus: entSchema.ActivityStatus(ss.Physical[tenant].Status),
```

Comment:
jus a suggestion, you can create a `createTenant` method that you could use here and in line 315 so that this part of code won't be copied